app/models/concerns/products/schedule_rule_support.rb

```diff
@@ -0,0 +1,36 @@
+module Products::ScheduleRuleSupport
+
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :schedule_rules, foreign_key: :product_id, inverse_of: :product
+  end
+
+  def can_purchase?(group_ids = nil)
+    if schedule_rules.empty?
```

Comment:
do we prefer the guard clause style? not sure if rubocop would find this `return false if schedule_rules.empty?`